core/services/directrequestocr/drlistener.go

```diff
@@ -251,11 +264,11 @@ func (l *DRListener) handleOracleRequest(request *ocr2dr_oracle.OCR2DROracleOrac
 		return
 	}
 
-	computationResult, errResult := l.jobORM.FindTaskResultByRunIDAndTaskName(run.ID, ParseResultTaskName)
+	computationResult, errResult := l.jobORM.FindTaskResultByRunIDAndTaskName(run.ID, ParseResultTaskName, pg.WithParentCtx(ctx))
 	if errResult != nil {
 		// Internal problem: Can't find parsed computation results
-		if err2 := l.pluginORM.SetError(request.RequestId, run.ID, NODE_EXCEPTION, []byte(errResult.Error()), time.Now()); err2 != nil {
-			l.logger.Errorw("call to SetError failed", "requestID", formatRequestId(request.RequestId))
+		if err2 := l.pluginORM.SetError(request.RequestId, run.ID, NODE_EXCEPTION, []byte(errResult.Error()), time.Now(), pg.WithParentCtx(ctx)); err2 != nil {
```

Comment:
SonarQube said this SetError pattern is repeated 3 times across the file with the only difference in error constant. Maybe it worth extracting this to a separate function?

Reply:
This will also eliminate the other SonarQube error:
`Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed.`

Reply:
I'm not a huge fan of this as this functions will have 5 parameters anyway so it's almost like calling SetError directly.